### PR TITLE
[cloud-provider-vsphere] Bootstrap to an existing vm folder

### PR DESCRIPTION
**Pull request by vitaliy-sn**
## Description

Added a parameter that allows you to use an existing vm folder.

## Why do we need it, and what problem does it solve?

Customers have requirements for the names of the directories where the cluster is installed and may not be granted rights in VMware vSphere to create a folder.

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: cloud-provider-vsphere
type: feature
summary: Added a parameter that allows you to use an existing vm folder.
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @vitaliy-sn. I've started the workflow for label 'e2e/run/vsphere'...

**Comment by vitaliy-sn** [comment]
/e2e/abort vsphere;Standard;containerd;1.25 8138882605 8138882605-1-con-1-25 /sys/deckhouse-oss/install:pr7667 ubuntu@10.220.203.105

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @vitaliy-sn. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @vitaliy-sn. I've started the workflow for label 'e2e/run/vsphere'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @vitaliy-sn. I've started the workflow for label 'e2e/run/vsphere'...

File: ee/candi/cloud-providers/vsphere/openapi/doc-ru-cluster_configuration.yaml [hunk 1]
```diff
@@ -176,6 +176,9 @@ apiVersions:
         vmFolderPath:
           description: |
             Путь до VirtualMachine Folder, в котором будут создаваться склонированные виртуальные машины.
+        vmFolderExists:
+          description: |
+            Установите значение в true, если путь заданный в vmFolderPath существует. Установка более одного кластера в в папку невозможна.
```
Comment (z9r5) [inline ee/candi/cloud-providers/vsphere/openapi/doc-ru-cluster_configuration.yaml:181]:
```suggestion
            Установите значение в `true`, если путь заданный в `vmFolderPath` существует. Установка более одного кластера в папку невозможна.
```

File: ee/candi/cloud-providers/vsphere/openapi/cluster_configuration.yaml [hunk 2]
```diff
@@ -360,6 +360,14 @@ apiVersions:
         x-examples:
           - "dev/test"
         x-unsafe: true
+      vmFolderExists:
+        type: boolean
+        default: false
+        description: |
+          Set the value to true if the path specified in vmFolderPath exists. Installing more than one cluster in a folder is not possible.
```
Comment (z9r5) [inline ee/candi/cloud-providers/vsphere/openapi/cluster_configuration.yaml:367]:
```suggestion
          Set the value to `true` if the path specified in `vmFolderPath` exists. Installing more than one cluster in a folder is not possible.
```

**Review by z9r5 — APPROVED** [review]

**Review by RomanenkoDenys — APPROVED** [review]

File: ee/candi/cloud-providers/vsphere/openapi/cluster_configuration.yaml [hunk 3]
```diff
@@ -360,6 +360,14 @@ apiVersions:
         x-examples:
           - "dev/test"
         x-unsafe: true
+      vmFolderExists:
+        type: boolean
+        default: false
+        description: |
+          Set the value to `true` if the path specified in `vmFolderPath` exists. Installing more than one cluster in a folder is not possible.
+        x-examples:
+          - "dev/test"
```
[the remark this answers is not in the document]
Comment (shvgn) [reply ee/candi/cloud-providers/vsphere/openapi/cluster_configuration.yaml:369]: @vitaliy-sn is this intended?

**Pull request merged**